### DISCUSSION TITLE
List of supported commands in Code Lenses

### DISCUSSION THREAD

**Discussion by khalilboussetta** [microsoft/vscode-jupyter#11201] [Questions and Answers] [answered]
Hi,

I am trying to add some commands to the Code Lenses parameters (e.g. jupyter.restartkernel and jupyter.interactive.clearAllCells) to have them on top of each cell but it doesn't show.

My Code Lenses JSON looks like this:

> "jupyter.codeLenses": "jupyter.runcell, jupyter.runallcellsabove, jupyter.debugcell, jupyter.interactive.clearAllCells, jupyter.restartkernel" 

Is there a list of accepted commands for the Code Lenses somewhere? or am I doing something wrong?

Thanks

**Comment by IanMatthewHuff** [comment, marked as answer]
Apologies, that setting is maybe a bit misnamed or confusing. It doesn't allow you to add any specific commands to the Code Lenses. It instead was added for customers who wanted only some of the standard code lenses provided. So you can have the initial default list, or a subset of that list, but you can't add new commands to it that were not there initially.
